openstacksdk_plugin/resources/compute/server.py

```diff
@@ -577,11 +669,231 @@ def _update_server_config_from_relationships(server_config):
+
+def _connect_resources_to_external_server(openstack_resource):
+    """
+    This method will try to connect resources to external server
+    :param openstack_resource: Instance Of OpenstackServer in order to
+    use it
+    """
+
+    # Try to connect networks to external server
+    _connect_networks_to_external_server(openstack_resource)
+
+    # Validate external key pair connected to the external server
+    _connect_keypair_to_external_server(openstack_resource)
+
+
+def _disconnect_resources_from_external_server(openstack_resource):
+    """
+    This method will disconnect networks from external server so that they
+    can be removed without any issue
+    :param openstack_resource: Instance Of OpenstackServer in order to
+    use it
+    """
+    # Delete all interfaces added to the external server
+    if SERVER_INTERFACE_IDS in ctx.instance.runtime_properties:
+        interfaces = ctx.instance.runtime_properties.get(
+            SERVER_INTERFACE_IDS, [])
+        for interface in interfaces:
+            openstack_resource.delete_server_interface(interface)
+            ctx.logger.info(
+                'Successfully detached network {0} to device (server) id {1}.'
+                .format(interface, openstack_resource.resource_id))
+
+
+def _assign_server_payload_as_runtime_properties(payload):
+    """
+    Store server configuration in the runtime
+    properties and cleans any potentially sensitive data.
+    :param payload: The payload.
+    """
+    if getattr(ctx, 'instance') and payload:
+        if SERVER_OPENSTACK_TYPE not in ctx.instance.runtime_properties.keys():
+            ctx.instance.runtime_properties[SERVER_OPENSTACK_TYPE] = {}
+        for key, value in payload.items():
+            if key not in ['user_data', 'adminPass']:
+                ctx.instance.runtime_properties[SERVER_OPENSTACK_TYPE][key]\
+                    = value
+
+
+def _get_user_password(openstack_resource):
+    """
+    This method will get the server password as encrypted for the current
+    server
+    :param openstack_resource: Instance Of OpenstackServer in order to
+    use it
+    """
+    if ctx.node.properties['use_password']:
+        # The current openstack sdk does not allow to send private key path
+        # when trying to lookup the password which means the password
+        # generated will be encrypted
+        res = openstack_resource.get_server_password()
+        password = json.loads(res.content) if res.content else None
+        password = password['password'] if password.get('password') else None
+        # If the password is not set then, again
+        if not password:
+            raise OperationRetry(
+                message='Waiting for server to post generated password')
+        else:
+            # Encrypted password, in order to decrypt it, decrypt it manually
+            ctx.instance.runtime_properties[SERVER_ADMIN_PASSWORD] = password
+            ctx.logger.info('Server has been set with a password')
```

Comment:
Idea for after release - to give option to create secret.

Reply:
I agree with you on this, but did you notice that the password generated cannot be decrypt because the openstack sdk does not allow to pass private key when getting the password which means it is up to user to decrypt it manually since he will has the private key